src/common/device_helpers.cuh

```diff
@@ -137,6 +137,21 @@ inline int get_device_idx(int gpu_id) {
   return (std::abs(gpu_id) + 0) % dh::n_visible_devices();
 }
 
+inline void check_compute_capability(){
```

Comment:
Check compute capability function.  Can have return actual number to check if (e.g.) >=6.0 for pascal to trigger different double precision operations.